src/commands/addMIConnections/addLocalMIConnections.ts

```diff
@@ -0,0 +1,58 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { AppSettingTreeItem } from "@microsoft/vscode-azext-azureappsettings";
+import { AzureWizard, type AzureWizardExecuteStep, type AzureWizardPromptStep, type IActionContext } from "@microsoft/vscode-azext-utils";
+import { type MessageItem } from "vscode";
+import { localize } from "../../localize";
+import { createActivityContext } from "../../utils/activityUtils";
+import { ConnectionsListStep, type Connection } from "./ConnectionsListStep";
+import { type IAddMIConnectionsContext } from "./IAddMIConnectionsContext";
+import { LocalSettingsAddStep } from "./LocalSettingsAddStep";
+import { SettingsAddBaseStep } from "./SettingsAddBaseStep";
+
+export async function addLocalMIConnections(context: IActionContext, node?: AppSettingTreeItem): Promise<void> {
+    const connections: Connection[] = [];
+    if (node instanceof AppSettingTreeItem) {
+        connections.push({ name: node.id, value: node.value });
+        await node.parent.runWithTemporaryDescription(context, localize('adding', 'Adding...'), async () => {
+            await addLocalMIConnectionsInternal(context, connections);
+        });
+    } else {
+        await addLocalMIConnectionsInternal(context);
+    }
+}
+
+export async function addLocalMIConnectionsInternal(context: IActionContext, connections?: Connection[]): Promise<void> {
+    const wizardContext: IActionContext & Partial<IAddMIConnectionsContext> = {
+        ...context,
+        ...await createActivityContext()
+    };
+
+    wizardContext.connections = connections;
+
+    wizardContext.activityChildren = [];
+
+    const title: string = localize('addLocalMIConnectionsInternal', 'Add Identity-Based Connections to Local Project');
+
+    const promptSteps: AzureWizardPromptStep<IAddMIConnectionsContext>[] = [];
+    const executeSteps: AzureWizardExecuteStep<IAddMIConnectionsContext>[] = [];
+
+    promptSteps.push(new ConnectionsListStep())
```

Comment:
nit: You're not pushing steps in dynamically anymore so might as well just instantiate promptSteps and executeSteps with the steps that they need instead of creating empty arrays first,